File: go/0257E-二叉树的所有路径.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-04 10:05:02
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */

package main

import (
	"strconv"
)

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func binaryTreePaths(root *TreeNode) []string {
	ans := make([]string, 0)
	var dfs func(node *TreeNode, res string)
	dfs = func(node *TreeNode, res string) {
		if node == nil {
			return
		}
		res += strconv.Itoa(node.Val)
		if node.Left == nil && node.Right == nil {
			ans = append(ans, res)
		} else {
			res += "->"
			dfs(node.Left, res)
			dfs(node.Right, res)
		}
	}
	dfs(root, "")
	return ans
}

File: cpp/0257E-二叉树的所有路径.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-21 10:31:08
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    vector<string> ans;

    void dfs(TreeNode* node, string tmp) {
        if (node == nullptr) return;
        if (node->left == nullptr && node->right == nullptr) {
            ans.push_back(tmp + to_string(node->val));
            return;
        }
        dfs(node->left, tmp + to_string(node->val) + "->");
        dfs(node->right, tmp + to_string(node->val) + "->");
    }

    vector<string> binaryTreePaths(TreeNode* root) {
        dfs(root, "");
        return ans;
    }
};